src/Aspire.Hosting.Maui/MauiOtlpExtensions.cs

```diff
@@ -0,0 +1,166 @@
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<T> WithOtlpDevTunnel<T>(
+        this IResourceBuilder<T> builder)
+        where T : IMauiPlatformResource, IResourceWithEnvironment
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        // Get shared state - only create stub + tunnel once per app
+        var platformResource = builder.Resource;
+        var parentBuilder = builder.ApplicationBuilder.CreateResourceBuilder(platformResource.Parent);
+        var configuration = builder.ApplicationBuilder.Configuration;
+
+        // Check if we already created the stub + tunnel for this MAUI project
+        if (!parentBuilder.Resource.TryGetLastAnnotation<OtlpDevTunnelConfigurationAnnotation>(out var tunnelConfig))
+        {
+            // First time - create stub and dev tunnel
+            tunnelConfig = CreateOtlpDevTunnelInfrastructure(parentBuilder, configuration);
+            parentBuilder.Resource.Annotations.Add(tunnelConfig);
+        }
+
+        // Now apply the configuration to this specific platform
+        ApplyOtlpConfigurationToPlatform(builder, tunnelConfig);
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Creates the OTLP dev tunnel infrastructure (stub resource + dev tunnel).
+    /// This is only created once per MAUI project and shared across all platforms.
+    /// </summary>
+    private static OtlpDevTunnelConfigurationAnnotation CreateOtlpDevTunnelInfrastructure(
+        IResourceBuilder<MauiProjectResource> parentBuilder,
+        Microsoft.Extensions.Configuration.IConfiguration configuration)
+    {
+        var appBuilder = parentBuilder.ApplicationBuilder;
+
+        // Resolve OTLP scheme and port from configuration
+        var (otlpScheme, otlpPort) = OtlpEndpointResolver.Resolve(configuration);
+
+        // Create names for the tunnel infrastructure
+        // Use a short random suffix to ensure uniqueness (similar to DCP naming strategy)
+        // The dev tunnel port resource name will be: {parent resource name}-{random}-otlp
+        var randomSuffix = Guid.NewGuid().ToString("N")[..8];
+        var tunnelName = parentBuilder.Resource.Name;
+        var stubName = $"t{randomSuffix}"; // Prefix with 't' to ensure valid resource name
+
+        // Create OtlpLoopbackResource - a synthetic IResourceWithEndpoints for service discovery
+        var stubResource = new OtlpLoopbackResource(stubName, otlpPort, otlpScheme);
+
+        var stubBuilder = appBuilder.AddResource(stubResource)
+            .ExcludeFromManifest();
+
+        // Hide the stub from the dashboard UI
+        stubBuilder.WithInitialState(new CustomResourceSnapshot
+        {
+            ResourceType = "OtlpStub",
+            Properties = [],
+            IsHidden = true
+        });
+
+        // Create dev tunnel with anonymous access for OTLP
+        var devTunnel = appBuilder.AddDevTunnel(tunnelName)
+            .WithAnonymousAccess()
```

Comment:
Defaulting to an anonymous tunnel isn't ideal here. We should consider making it more obvious that this is anonymous (method name change) or use the interaction service to force the user to confirm that they want to create the tunnel with anonymous access, and if they don't accept it, don't start the tunnel.